### ISSUE TITLE
Build error: #include "faust/dsp/libfaust.h"

### ISSUE THREAD

**Issue by Bubobubobubobubo** [madskjeldgaard/faustgen-supercollider#1]
Hi! I ended up with a compilation error that you might want to investigate. I am using macOS Big Sur version 11.5.2. The system is up to date, to the best of my knowledge:
* `supercollider` and `faustgen` sources are stored in my `root` directory.
* the first step worked as intended. The error is encountered when running `cmake --build .  --config Release` from the `build` directory.

```
[ 96%] Building CXX object CMakeFiles/FaustGen_supernova.dir/plugins/FaustGen/FaustGen.cpp.o
In file included from /Users/bubo/faustgen/plugins/FaustGen/FaustGen.cpp:4:
/Users/bubo/faustgen/plugins/FaustGen/FaustGen.hpp:7:10: fatal error: 'faust/dsp/libfaust.h' file not found
#include "faust/dsp/libfaust.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/FaustGen_supernova.dir/build.make:82: CMakeFiles/FaustGen_supernova.dir/plugins/FaustGen/FaustGen.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:241: CMakeFiles/FaustGen_supernova.dir/all] Error 2
gmake: *** [Makefile:171: all] Error 2
```

I do not use Cmake very often. It might be something trivial that I overlooked.

**Comment by madskjeldgaard** [comment]
Hmm. Interesting. It can't find the Faust source. Did you run the submodule update command as described in the readme ?


git submodule update --init --recursive


This should download a folder called Faust within the project.
Get Outlook for Android

**Comment by Bubobubobubobubo** [comment]
I runned `git submodule update --init --recursive`. The `faust` folder is here but does not contain any `dsp` folder.

There is a `faustgen.h` inside the `faust/compiler/generator` folder.

**Comment by madskjeldgaard** [comment]
Ah then it has something to do with where Faust is installed on your system. Did you install Faust before running this ? I might have to tweak cmakelists to better pick this up on macos

**Comment by Bubobubobubobubo** [comment]
Faust is installed on my system, version `2.28.8`. The command `which faust` returns `/usr/local/bin/faust` but isn't the build process using the freshly cloned submodule?

I'll try to uninstall Faust to see if it changes something.

**Comment by madskjeldgaard** [comment]
> Faust is installed on my system, version `2.28.8`. The command `which faust` returns `/usr/local/bin/faust` but isn't the build process using the freshly cloned submodule?
> 
> I'll try to uninstall Faust to see if it changes something.

Yeah you're right. As far as I can see, the CMake setup as it is now should pick this up if the submodule has been downloaded correctly. Hmm.

**Comment by Bubobubobubobubo** [comment]
@madskjeldgaard : tried again this morning but stumbled upon the same error as two years ago. 

```
/Users/bubo/faustgen/plugins/FaustGen/FaustGen.hpp:7:10: fatal error: 'faust/dsp/libfaust.h' file not found
#include "faust/dsp/libfaust.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/FaustGen_scsynth.dir/build.make:76: CMakeFiles/FaustGen_scsynth.dir/plugins/FaustGen/FaustGen.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:139: CMakeFiles/FaustGen_scsynth.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

**Comment by madskjeldgaard** [comment]
Thanks. I'm not working on this anymore or at the moment at least, but I think this might be solved by having CMake manage Faust as a dependency using CPM, sort of how PortedPlugins does it https://github.com/madskjeldgaard/portedplugins/blob/main/CMakeLists.txt
